gigascience/gigadb-website
# Remove local Bootstrap styles and use Bootstrap from CDN only

# Pull request for issue: #1356

This is a pull request for the following functionalities:

* Get rid of local bootstrap files
* Fix styles broken as a result of removing bootstrap files

## How to test?

Deploy frontend locally, access admin pages and see that there are no broken styles

## How have functionalities been implemented?

Front uses two different sets of styles which have as entry points:

- current.less (for new layouts)
- site.less (for old layouts).

site.less imported local bootstrap files that now have been removed.

site.less is used as stylesheet in admin pages, so it was necessary to fix what was broken when local bootstrap was removed. The fix involves:

- Loading bootstrap 2.0.4 from CDN (the same version of the local files removed)
- Adding a small number of less variables and styles in less/manual-bootstrap-overrides.less to prevent errors on less compilation, and to fix a small number of very specific styling issues

Note that this still means the site uses two different bootstrap versions depending on the layout, so whereas we get rid of the local bootstrap files, this PR does not do much more than shifting the issue around.

I tried to use the same version of bootstrap used in the modern layout, but that completely breaks the styles.

I believe the next steps are:

- All pages use newer layouts (new_main.php, new_datasetpage.php)
- Update admin pages to the newer style (#1358 )
- Eventually delete old layouts that by this point will not be used anywhere
- Possibly rename new_main.php to main.php so that Yii uses it as default layout

## Any issues with implementation?

When I run ./up.sh, the composer.lock file gets some version updates. I suspect this should not happen as the last change to this file looks like from 3 months ago but I'm not familiar with composer. Please, let me know if I have to revert the composer.lock file

## Any changes to automated tests?

None

## Any changes to documentation?

None

## Any technical debt repayment?

None

## Any improvements to CI/CD pipeline?

None
